Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1479,10 +1505,18 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
         CheckedPtr<RenderBox> gridItem = it.get();
         accumulateIntrinsicSizes(gridItem.get());
     }
+
+    if (track.cachedTrackSize().minTrackBreadth().isFlex() || track.cachedTrackSize().maxTrackBreadth().isFlex()) {
+        for (auto it = singleTrackMasonryIndefiniteItems.begin(), end = singleTrackMasonryIndefiniteItems.end(); it != end; ++it) {
+            CheckedPtr<RenderBox> gridItem = it.get();
+            accumulateIntrinsicSizes(gridItem.get());
+        }
+    }
```

Comment:
This logic seems to kinda come out of nowhere and it took me a bit to understand what its intent was in terms of the optimization/patch. Correct me if I am wrong, but is this to handle the `In case we hit an 'fr' track we will fallback to computing all indefinite items in the track` portion of your comment? 

If so, can we find a way to restructure the code so that we do not even attempt this optimization at all if we find some sort of content that we currently don't support? I think that would really improve the readability of the code instead of trying to interleave logic that tries to put us back into a state as if we did not try to do any sort of optimization.

Reply:
Exactly, if we hit an 'fr' track we can't apply the optimization.